modules/virt-retrieving-vm-definition-from-vme-manifest.adoc

```diff
@@ -0,0 +1,72 @@
+// Module included in the following assemblies:
+//
+// * virt/virtual_machines/virt-exporting-vms.adoc
+
+:_content-type: PROCEDURE
+[id="virt-retrieving-vm-definition-from-vme-manifest_{context}"]
+= Retrieving a VM definition from a virtual machine export manifest
+
+After you export a virtual machine (VM) or snapshot, you can access the `VirtualMachine` manifest and related information from the export server.
+
+.Prerequisites
+
+* You exported a virtual machine or VM snapshot by creating a `VirtualMachineExport` custom resource (CR).
++
+[NOTE]
+====
+`VirtualMachineExport` objects that have `spec.source.kind: PersistentVolumeClaim` do not generate virtual machine manifests.
+====
+
+.Procedure
+
+. Get the `VirtualMachineExport` custom resource by running the following command:
++
+[source,terminal]
+----
+$ oc get vmexport <export_name> -o yaml <1>
+----
+<1> Replace `<export_name>` with the `metadata.name` value from the `VirtualMachineExport` object.
+
+. Review the `status.links` stanza, which is divided into `external` and `internal` sections. Note the `manifests.url` fields within each section:
++
+.Example output
+
+[source,yaml]
+----
+apiVersion: export.kubevirt.io/v1alpha1
+kind: VirtualMachineExport
+metadata:
+  name: example-export
+spec:
+  source:
+    apiGroup: "kubevirt.io"
+    kind: VirtualMachine
+    name: example-vm
+  tokenSecretRef: example-token
+status:
+...
+  links:
+    external:
+...
+      manifests:
+      - type: all
+        url: https://vmexport-proxy.test.net/api/export.kubevirt.io/v1alpha1/namespaces/example/virtualmachineexports/example-export/external/manifests/all <1>
+      - type: auth-header-secret
+        url: https://vmexport-proxy.test.net/api/export.kubevirt.io/v1alpha1/namespaces/example/virtualmachineexports/example-export/external/manifests/secret <2>
+    internal:
+...
+      manifests:
+      - type: all
+        url: https://virt-export-export-pvc.default.svc/internal/manifests/all <3>
+      - type: auth-header-secret
+        url: https://virt-export-export-pvc.default.svc/internal/manifests/secret
+  phase: Ready
+  serviceName: virt-export-example-export
+----
+<1> Access the `VirtualMachine` manifest, `DataVolume` objects, if present, and a config map that contains the public certificate for the external URL's ingress or route.
+<2> Access a secret containing a header that is compatible with Containerized Data Importer (CDI). The header contains a text version of the export token.
+<3> Access the `VirtualMachine` manifest, `DataVolume` objects, if present, and a config map that contains the certificate authority (CA) of the internal URL's export server.
+
```

Comment:
I'm just wondering if we should add the instructions on how to get the manifests from the manifests links, because it requires having the certificates in the target cluster.
I'm using these steps:

Source cluster: save the certificates. 
Target cluster: copy cacert.crt to the target cluster
`oc get vmexport example-export -o jsonpath={.status.links.external.cert} > cacert.crt
`

Source cluster: decode the token and save it:
`oc get secret export-token-example-export -o jsonpath={.data.token} | base64 --decode > token_decode
`

Target cluster: get secret headers yaml using a secret manifest and a `token_decode` from the source cluster. Create the Secret

`curl --cacert cacert.crt https://virt-exportproxy-openshift-cnv.test.net/api/export.kubevirt.io/v1alpha1/namespaces/default/virtualmachineexports/example-export/external/manifests/secret -H "x-kubevirt-export-token:<token_decode>" -H "Accept:application/yaml"
`

Reply:
This sounds really important to include; thank you! I'm a little confused about how I should sequence this, though. Am I understanding this correctly? Thanks so much @jpeimer 

1. Copy cacert.crt to the target cluster (by running the command you provided from the target cluster, IIUC)
2. Decode the token and save it (by running the command you provided from the source cluster)
3. Existing step: get the custom resource
4. Existing step: examine the CR and identify the manifest URLs
5. From the target cluster, create the Secret by running the `curl --cacert ...` command, referencing a Secret manifest URL (that you got from the YAML in step 4) and the token_decode (that you created in step 2)
6. Access the manifest URLs to your heart's content and/or apply them to the cluster